lib/client.js

```diff
@@ -352,20 +352,10 @@ export default class Client {
       this.ready = true
       this._metadata = data.metadata
       this._context = data.context
-      this._iframe_session_timeout = data.iframe_session_timeout
-      this._idleState = new IdleState(data.iframe_session_timeout)
-      // Add an observer.  This will be called when the idleState changes
-      this._idleState.addObserver(state => {
-        if (state === 'active') {
-          // ... application is now active, timeout clock is reset to zero
-          // Send a postMessage indicating liveness + possibly a payload too
-          this.postMessage('session.live')
-        } else if (state === 'idle') {
-          // ... application is now idle, timeout clock is ticking
-          // Send a postMessage indicating sessionTimeout + possibly a payload too
-          this.postMessage('session.idle')
-        }
-      })
+      this._idleState = new IdleState()
+      this._idleState.addObserver(debounce(() => {
+        this.postMessage('session.live')
```

Comment:
A comment explaining why we used a debounce may be helpful here:
```suggestion
        /**
         * Sends an indication that the user is active in the iframe.
         * Adding a debounce avoids a bombardment of unnecessary postMessages every microsecond (like when a user drags their mouse).
         */
        this.postMessage('session.live')
```